datajoint_utilities/typing/__init__.py

```diff
@@ -0,0 +1,180 @@
+from datajoint_utilities.generic.pkgutil import detect_min_python_version
+
+detect_min_python_version(3, 10)
+
+import datetime
+import inspect
+import types
+from pathlib import Path
+from typing import (
+    Any,
+    ItemsView,
+    Literal,
+    Mapping,
+    MutableMapping,
+    Protocol,
+    Sequence,
+    Sized,
+    TypeAlias,
+    TypedDict,
+    TypeGuard,
+    TypeVar,
+    cast,
+    runtime_checkable,
+)
+from uuid import UUID
+
+import datajoint as dj
+from numpy import bool_, complex_, float_, int_, ndarray, str_
+from numpy.typing import NDArray
+
+# Misc. typing -------------------------------------------------------------------------
+T = TypeVar("T")
+DictObj: TypeAlias = dict[str, object]
+DictAny: TypeAlias = dict[str, Any]
+AnyMap: TypeAlias = Mapping[str, object]
+AnyMutMap: TypeAlias = MutableMapping[str, object]
+StrNone: TypeAlias = str | None
+StrPath: TypeAlias = str | Path
+StrPathNone: TypeAlias = str | Path | None
+StrTuple: TypeAlias = tuple[str, ...]
+StrPathParts: TypeAlias = str | Path | StrTuple
+FrameType: TypeAlias = types.FrameType
+FrameInfo: TypeAlias = inspect.FrameInfo
+FrameInfoList: TypeAlias = list[FrameInfo]
+FrameStack: TypeAlias = FrameInfoList | FrameType
+FrameFile: TypeAlias = str | Path | FrameInfo
+FrameFileNone: TypeAlias = str | Path | FrameInfo | None
+AnyMapMapSeq: TypeAlias = AnyMap | Sequence[AnyMap]
+AnyMapStrSeq: TypeAlias = AnyMap | Sequence[str] | str
+TimeStampable: TypeAlias = datetime.datetime | datetime.date | datetime.timedelta
+UUIDStr: TypeAlias = str | UUID
+UUIDLike: TypeAlias = (
+    str | UUID | MutableMapping[str, str | UUID] | dj.expression.QueryExpression
+)
+ModuleType: TypeAlias = types.ModuleType
+
+
+def is_pathstr(obj: object) -> TypeGuard[StrPath]:
+    return isinstance(obj, (Path, str))
+
+
+def is_frame(obj: object) -> TypeGuard[FrameType]:
+    """Determines whether and object is a frame from inspect.currentframe."""
+    return isinstance(obj, types.FrameType)
+
+
+def is_frame_info(obj: object) -> TypeGuard[FrameInfo]:
+    """Determines whether and object is a frame from inspect.stack list."""
+    return isinstance(obj, inspect.FrameInfo)
+
+
+def is_frame_info_list(obj: object) -> TypeGuard[FrameInfoList]:
+    """Determines whether and object is a stack list inspect.stack."""
+    if not isinstance(obj, list):
+        return False
+    f: Any
+    for f in obj:
+        if not is_frame_info(f):
+            return False
+    return True
+
+
+def is_frame_stack(obj: object) -> TypeGuard[FrameStack]:
+    return is_frame(obj) or is_frame_info_list(obj)
+
+
+def is_strmap(obj: object, allow_empty: bool = True) -> TypeGuard[AnyMap]:
+    if not isinstance(obj, Mapping):
+        return False
+    if len(cast(Sized, obj)) == 0:
+        return allow_empty
+    return all(
+        isinstance(k, str) and isinstance(v, object)
```

Comment:
Everything is an instance of object
```suggestion
        isinstance(k, str)
```